04week/mastermind.js

```diff
@@ -28,50 +28,84 @@ function getRandomInt(min, max) {
   return Math.floor(Math.random() * (max - min)) + min;
 }
 
-function generateHint() {
-  // your code here
+function generateHint(guess) {
+  let solutionArray = solution.split("");
+  // console.log(solutionArray);
+  let guessArray = guess.split("");
+  // console.log(guessArray);
+
+  // +++++++ Counter for Red Hint ++++++++
+  var correctLetterLocations = 0;
+
+  for (let i = 0; i < solutionArray.length; i++) {
+    if (solutionArray[i] === guessArray[i]) {
+      correctLetterLocations++;
+      solutionArray[i] = null;
+    }
+  }
+
+  // +++++++ Counter for White Hint ++++++++
+  var correctLetters = 0;
+
+  for (let i = 0; i < solutionArray.length; i++) {
+    var targetIndex = solutionArray.indexOf(guessArray[i]);
+    if (targetIndex > -1) {
+      correctLetters++;
+      solutionArray[i] = null;
```

Comment:
you should null out the solutionArray at the targetIndex, since that was where the letter was.